RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+        public Tile[] Shuffle()
+        {
+            // Shuffle the tiles randomly
+            Random random = new Random();
+            for (int i = 0; i < tiles.Length; i++)
+            {
+                int j = random.Next(i, tiles.Length);
+                Tile temp = tiles[i];
+                tiles[i] = tiles[j];
+                tiles[j] = temp;
+            }
+            return tiles;
+        }
+
+       
+        public Tile[] Deal()
+        {
+            // Deal 14 tiles from the deck to start the game
+            for (int i = 0; i < 14; i++)
+            {
+                hand[i] = tiles[i];
+            }
+            return hand;
+        }
+        public Tile[] DealAI()
+        {
+            // Deal 14 tiles from the deck to start the game
+            for (int i = 0; i < 14; i++)
+            {
+                AIhand[i] = tiles[i+14];
+            }
+            return hand;
+        }
+
+
+        public Tile[] bynumber()
+        {
+            int count = 0;
+            while (hand[count] != null)
+                count++;
+            for (int i = 0; i < count - 1; i++)
+                for (int j = 0; j < count - i - 1; j++)
+                    if (hand[j].Value > hand[j + 1].Value)
+                    {
+                        // swap temp and arr[i]
+                        Tile temp = hand[j];
+                        hand[j] = hand[j + 1];
+                        hand[j + 1] = temp;
+                    }
+            return hand;
+        }
+
+        public Tile[] bycolor()
+        {
+            hand = bynumber();
+            int n = 0;
+            while (hand[n] != null)
+                n++;
+            int count = 0;
+            Tile[] temp = new Tile[n];
+            for (int i = 0; i < n; i++)
+            {
+                if (hand[i].Color == "Yellow")
+                {
+                    temp[count] = hand[i];
+                    count++;
+                }
+            }
+            for (int i = 0; i < n; i++)
+            {
+                if (hand[i].Color == "Blue")
+                {
+                    temp[count] = hand[i];
+                    count++;
+                }
+            }
+            for (int i = 0; i < n; i++)
+            {
+                if (hand[i].Color == "Red")
+                {
+                    temp[count] = hand[i];
+                    count++;
+                }
+            }
+            for (int i = 0; i < n; i++)
+            {
+                if (hand[i].Color == "Black")
+                {
+                    temp[count] = hand[i];
+                    count++;
+                }
+            }
+            for (int i = 0; i < n; i++)
+            {
+                if (hand[i].Color == "Joker")
+                {
+                    temp[count] = hand[i];
+                    count++;
+                }
+            }
+            return temp;
+        }
+
+
+        private int count = 0;
+        public Tile AddTileToHand()

```

Comment:
In AddTileToHand, is it possible for someone to have more than 14 tiles in hand? if I have 14 tiles in hand wil the while loop crash? maybe use List object instead of array for hand management.